salt/modules/file.py

```diff
@@ -3740,16 +3740,22 @@ def _get_local_file_source_sum(path):
 
     if source and (template or parsed_scheme in salt.utils.files.REMOTE_PROTOS):
         # Check if we have the template or remote file cached
+        cache_refetch = False
         cached_dest = __salt__['cp.is_cached'](source, saltenv)
         if cached_dest and (source_hash or skip_verify):
             htype = source_sum.get('hash_type', 'sha256')
             cached_sum = get_hash(cached_dest, form=htype)
-            if skip_verify or cached_sum == source_sum['hsum']:
+            if cached_sum != source_sum['hsum']:
```

Comment:
It seems that when `skip_verify == true`, the `hsum` would be not present in `source_sum`, leading to:

```Python traceback
[DEBUG   ] Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/salt/states/file.py", line 1836, in managed
    **kwargs
  File "/usr/lib/python2.7/site-packages/salt/modules/file.py", line 3748, in get_managed
    if cached_sum != source_sum['hsum']:
KeyError: 'hsum'
```